.github/scripts/validate_binaries.sh

```diff
@@ -102,6 +102,7 @@ else
     ${PYTHON_RUN}  ./smoke_test/smoke_test.py ${TEST_SUFFIX}
     # For pip install also test with latest numpy
     if [[ ${MATRIX_PACKAGE_TYPE} == 'wheel' ]]; then
+        # test with latest numpy 2.x
         pip3 install numpy --force-reinstall
```

Comment:
Suggest adding a `--upgrade` param to make it more future proof. https://github.com/pypa/pip/issues/8238

```suggestion
        pip3 install numpy --upgrade --force-reinstall
```